server/stycoon/command_trade.go

```diff
@@ -0,0 +1,35 @@
+package stycoon
+
+import (
+	"fmt"
+
+	"github.com/gdg-garage/space-tycoon/server/database"
+)
+
+func NewTradeCommand(command Command) *TradeCommand {
+	return &TradeCommand{
+		Type:     command.Type,
+		Amount:   command.Amount,
+		Resource: command.Resource,
+		Target:   command.Target,
+	}
+}
+
+func (game *Game) processTrade(id int64, playerId int64, command Command) error {
+	c := NewTradeCommand(command)
+	err := AssertTradeCommandRequired(*c)
+	if err != nil {
+		return err
+	}
+
+	targetIsShip, owner, err := database.CheckIfObjectIsShipAndGetOwner(game.db, id)
```

Comment:
use `playerOwnedShips` to avoid unnecessary sql queries here too

Reply:
done - checking the `playerOwnedShips` first and querying the db only if it's not found there